Add tsam to datapackage

iuljan opened the pull request:
# Description

Tsam option is now available for tabular reading module, but only for multi-period approach.
Parameters need to be provided by two separate `csv`s. 
- tsa_paramters (period, timesteps_per_period, order)

- [x] add example
- [x] adjust timeseries data of example

Fixes #138

## Type of change

Please tick or delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

Please tick or delete options that are not relevant.

- [x] New and adjusted code is formatted using the `pre-commit` hooks
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added new features/fixes to the [CHANGELOG](https://github.com/oemof/oemof-tabular/tree/dev/CHANGELOG.rst)

Commit:
Merge branch 'dev' into features/add-tsam-to-datapackage

Commit:
Add tsam to reading module

Tsam is currently only available for multi-period.

Commit:
Add tsam multi-period example

henhuy reviewed (commented):
Structure in datapackage looks good to me!

But I have failing tests on my side:
- test_example_datapackage_readability
- test_scripting_examples
- test_examples_datapackages_scripts_infer

Do the tests also fail on your side?

henhuy commented:
Why did yo mark it as "breaking change"? IMO, this should not interfere with older tabular versions?!

iuljan commented:
> Why did yo mark it as "breaking change"? IMO, this should not interfere with older tabular versions?!

I guess, I slipped...

> Do the tests also fail on your side?

yes, I still have to adjust the component timeseries.

Commit:
Remove timesteps from tsa_parameters

TSA timesteps are now disaggregated internally from periods
in oemof.solph

Commit:
Adjust timeseries data for tsam example

Commit:
Remove timeindex.csv

Commit:
Add changelog entry

Commit:
Remove timeindex.csv from datapackage.json

Commit:
Add tsam resource to infer_metadata

Commit:
Merge branch 'dev' into features/add-tsam-to-datapackage

iuljan committed:
Merge branch 'dev' into features/add-tsam-to-datapackage

henhuy reviewed (changes requested):
When looking into TSAm parameters of example datapackage "dispatch_multi_period_tsam", I found that third period has an invalid order (containing 3 different typical periods, which would sum up to 6 timesteps in third period, but only 4 are given) which must normally lead to an error when building energysystem. Thus, I wondered why no error is thrown - on my side, this was due to wrong `oemof.solph` version (one which does not include TSAM parameters - so that they are neglected when building ES ergo no error is thrown). Updating `oemof.solph`, pointing to branch `feature/integrate_tsam` unfortunately lead to dependency issues as `oemof.network`and `oemof.tools` versions are not correctly pinned in `oemof.solph`. After manually fixing those versions, I finally could see the error resulting from invalid TSAM parameters - but unfurtunately, now I can see multiple other errors in other tests (regarding wrong LP-files and stuff).
I think, in order to finalize this PR, we should do following:
- [x] fix dependencies in `oemof.solph`
- [x] fix TSAM parameters in example
- [x] look into errors from other tests and check if they belong into this PR or have to be adapted in other PR (as bringing `oemof.tabular` in line with latest `oemof.solph` version)

Commit:
Add further exceptions for not converted periodic values

Commit:
Make lifetime parameter a constant value

Commit:
Raise error if length mismatches periods or index

Commit:
Merge branch 'features/add-tsam-to-datapackage' of github.com:oemof/oemof-tabular into features/add-tsam-to-datapackage

FelixMau committed:
tsam period order 2-0

FelixMau committed:
Solph dependency

FelixMau committed:
Solph dependency

FelixMau committed:
Sepcify oemof.solph and oemof.network

FelixMau commented:
Regarding changed LP files please see #155

FelixMau committed:
Merge remote-tracking branch 'origin/features/add-tsam-to-datapackage' into features/add-tsam-to-datapackage

# Conflicts:
#	setup.py

FelixMau committed:
Merge branch 'dev' into features/add-tsam-to-datapackage

# Conflicts:
#	setup.py
#	src/oemof/tabular/datapackage/reading.py

FelixMau committed:
Linting

iuljan commented:
This will only be merged after [tsam-branch](https://github.com/oemof/oemof-solph/pull/980) in `solph` is merged.

henhuy committed:
Add option to set up TSAM Model without multi period mode

henhuy committed:
Read TSAM parameters for all timeindex options

henhuy committed:
Update datapackage and tableschema to latest versions

henhuy committed:
Fix foreign keys in reading of datapackage for latest datapackage version

henhuy committed:
Fix duplicate columns in example datapackages

This leads to an error when inferring metadata, as new datapackage adds a counter to second "tech" column, resulting in "tech2"

henhuy committed:
Check and fix LP files for extraction investment datapackages

henhuy committed:
Update python version in setup.py

henhuy committed:
Make linter happy

henhuy committed:
Allow older versions of datapackage and tableschema

henhuy committed:
Add changes to CHANGELOG.rst

henhuy committed:
Update python version for GitHub Actions

henhuy committed:
Merge 'feature/update_datapackage_version'

henhuy committed:
Fix typo in parameter name from 'tsam_parameters' to 'tsa_parameters'
